pyartifactory/objects.py

```diff
@@ -1,684 +1,745 @@
+            logging.debug(f"User {username} successfully created")
+            return self.get(user.name)
+
+    def get(self, name: str) -> UserResponse:
+        """
+        Read user from artifactory. Fill object if exist
+        :param name: Name of the user to retrieve
+        :return: UserModel
+        """
+        try:
+            r = self._get(f"api/{self._uri}/{name}")
+            return UserResponse(**r.json())
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == 404 or e.response.status_code == 400:
+                logging.error(f"User {name} does not exist")
+                raise UserNotFoundException(f"{name} does not exist")
+            raise ArtifactoryException from e
+
+    def list(self) -> List[SimpleUser]:
+        """
+        Lists all the users
+        :return: UserList
+        """
+        r = self._get(f"api/{self._uri}")
+        logging.debug("List all users successful")
+        return [SimpleUser(**user) for user in r.json()]
+
+    def update(self, user: User) -> UserResponse:
+        """
+        Updates an artifactory user
+        :param user: NewUser object
+        :return: UserModel
+        """
+        username = user.name
+        self.get(username)
+        self._post(f"api/{self._uri}/{username}", json=user.dict())
+        logging.debug(f"User {username} successfully updated")
+        return self.get(username)
+
+    def delete(self, name: str) -> None:
+        """
+        Remove user
+        :param name: Name of the user to delete
+        :return: None
+        """
+        self.get(name)
+        self._delete(f"api/{self._uri}/{name}")
+        logging.debug(f"User {name} successfully deleted")
+
+
+class ArtifactorySecurity(ArtifactoryObject):
+    _uri = "security"
+
+    def __init__(self, artifactory: AuthModel) -> None:
+        super(ArtifactorySecurity, self).__init__(artifactory)
+
+    def get_encrypted_password(self) -> PasswordModel:
+        """
+        Get the encrypted password of the authenticated requestor.
+        :return: str
+        """
+        response = self._get(f"api/{self._uri}/encryptedPassword")
+        logging.debug("Encrypted password successfully delivered")
+        return PasswordModel(**response.json())
+
+    def create_access_token(
+        self,
+        user_name: str,
+        expires_in: int = 3600,
+        refreshable: bool = False,
+        groups: list = None,
+    ) -> AccessTokenModel:
+        """
+        Creates an access token.
+
+        :param user_name: Name of the user to whom an access key should be granted. transient token created if not set.
+        :param expires_in: Expiry time for the token in seconds. For eternal tokens specify 0.
+        :param refreshable: If set to true token can be refreshed using the refresh token returned. defaults False.
+        :param groups: A list of groups the token has membership of. If username is specified, groups are implied.
```

Comment:
How can the `user_name` not be defined if it doesn't have a default value in the parameters?

Also what does "If username is specified, groups are implied." mean? I don't see any special behavior in the code for this case. The Artifactory docs say that if you specify `member-of-groups:*` it will use the defined user group, is this what you wanted to do?

Reply:
let me clarify here and I'll update the docs to match. 

if you create a token for a user that does exist in artifactory (and existing named user) you token will be created and a member of the groups that match the user. 

if you create a user that does NOT exist, the user will be 'transient' and created on the fly and groups are expected. If you create a transient user and you specify no groups at all, this will error with the following.

`token = art.security.create_access_token('non_existing_user')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/me/code/python/python-artifactory-fork/pyartifactory/objects.py", line 246, in create_access_token
    response.json().get("error_description", "Unknown error")
pyartifactory.exception.InvalidTokenDataException: Insufficient scope: 'api:*'`

now when you take an existing user which has existing group memberships, I can do this 

`token = art.security.create_access_token('docker-923-upload')
 >>>token.scope
'member-of-groups:docker-923-upload api:*'`

the group name that is returned is the group the actual existing user is a member of. 

I've rewritten the params to try clarify that.